subsys/bluetooth/host/gatt.c

```diff
@@ -4761,6 +4761,15 @@ static void parse_read_by_uuid(struct bt_conn *conn,
 		uint16_t handle;
 		uint16_t len;
 
+		len = MIN(rsp->len, length);
+		if (len < sizeof(struct bt_att_data)) {
+			LOG_WRN("Bad peer: ATT read-by-uuid rsp: invalid ATTR data len %u", len);
+			params->func(conn, BT_ATT_ERR_INVALID_ATTRIBUTE_LEN, params, NULL, 0);
```

Comment:
Why not `BT_ATT_ERR_INVALID_PDU`? Doesn't this mean that ATT PDU is malformed?

It is even a question if we are allowed to use response error codes here as it may look like server's response error. But doesn't seem that spec defines behavior for such case unless I'm missing something.

Reply:
Regardless of peer's intention, the ATT PDU is not correctly formed. `struct bt_att_data` is 2 bytes long. If the response has 1 byte left (half of ATT opcode), is this invalid attribute length or PDU?

Reply:
Corrected myself: ATT opcode -> attribute handle.

Reply:
Given the definitions in the Core spec

<img width="952" height="48" alt="image" src="https://github.com/user-attachments/assets/58909c30-ea6a-4404-b7ed-db3d0e17e263" />

<img width="954" height="76" alt="image" src="https://github.com/user-attachments/assets/6fe31279-0233-4510-b52b-4a72620f0a4e" />

I'd argue that "Invalid PDU" matches this case the best. I think the latter works better if you are doing a write request with invalid length